packages/react-components/src/components/NumberInput/NumberInput.jsx

```diff
@@ -13,12 +14,20 @@ class NumberInput extends React.Component {
   constructor() {
     super();
     this.handleChange = this.handleChange.bind(this);
+    this.state = { lang: null };
   }
 
   UNSAFE_componentWillMount() {
     this.inputId = uniqueId('errorable-number-input-');
   }
 
+  componentDidMount() {
+    i18next.on('languageChanged', lang => {
+      // We're using this state change to force a re-render
+      this.setState({ lang });
+    });
+  }
```

Comment:
We might want this component to unsubscribe from listening to this event on unmount. I was getting React errors about updating state on an unmounted component when switching the language attribute and switching between stories on my Select/va-select work.

https://github.com/department-of-veterans-affairs/component-library/pull/389/files#diff-e5a660d7959836c2a8f311d102223c28185e03dfdbbfe5b21a8758b41c60e75cR27-R29

https://www.i18next.com/overview/api#events